File: client/src/app/shared/models/audit.ts
import { PaginatedFilter } from "./pagination";

export class Audit {
  id: string;
  userId: string;
  type: string;
  tableName: string;
  timeStamp: string;
  oldValues: string;
  newValues: string;
  affectedColumns: string;
  primaryKey: string;
}

export class AuditParams implements PaginatedFilter {
  searchString: string;
  pageNumber: number = 1;
  pageSize: number = 5;
  orderBy: string;
}

File: hybrid/lib/entities/audit.dart
import 'package:json_annotation/json_annotation.dart';
part 'audit.g.dart';

@JsonSerializable()
class Audit {
  int id;
  int userId;
  String? type;
  String? tableName;
  DateTime dateTime;
  String? oldValues;
  String? newValues;
  String? affectedColumns;
  String? primaryKey;

  Audit(
    this.id,
    this.userId,
    this.type,
    this.tableName,
    this.dateTime,
    this.oldValues,
    this.newValues,
    this.affectedColumns,
    this.primaryKey,
  );

  factory Audit.fromJson(Map<String, dynamic> json) => _$AuditFromJson(json);

  Map<String, dynamic> toJson() => _$AuditToJson(this);
}
